src/semgrep_agent/ignores.py

```diff
@@ -1,312 +0,0 @@
-    cache: Dict[Path, Entry]
-
-    def __len__(self) -> int:
-        return len(self.cache)
-
-    def __iter__(self) -> Iterator[Entry]:
-        return iter(self.cache.values())
-
-    def __contains__(self, item: object) -> bool:
-        return isinstance(item, Entry) and item.path in self.cache
-
-
-@attr.s
-class FileIgnore(Mapping[Path, Entry]):
-    base_path = attr.ib(type=Path)
-    patterns = attr.ib(type=Set[str])
-    target_paths = attr.ib(type=List[Path])
-    _processed_patterns = attr.ib(type=Set[str], init=False)
-    _walk_cache: Dict[Path, Entry] = attr.ib(default=None, init=False)
-
-    def __attrs_post_init__(self) -> None:
-        self._processed_patterns = Processor(self.base_path).process(self.patterns)
-        debug_echo("Initializing file tree")
-        self._init_cache()
-
-    def _survives(self, path: Path) -> bool:
-        """
-        Determines if a single Path survives the ignore filter.
-        """
-        for p in self._processed_patterns:
-            if path.is_dir() and p.endswith("/") and fnmatch.fnmatch(str(path), p[:-1]):
-                return False
-            if fnmatch.fnmatch(str(path), p):
-                return False
-
-            # Check any subpath of path satisfies a pattern
-            # i.e. a/b/c/d is ignored with rule a/b
-            # This is a hack. TargetFileManager should be pruning while searching
-            # instead of post filtering to avoid this
-            # Note: Use relative to base to avoid ignore rules firing on parent directories
-            # i.e. /private/var/.../instabot should not be ignored with var/ rule
-            # in instabot dir as base_path
-            # Note: Append "/" to path before running fnmatch so **/pattern matches with pattern/stuff
-            if p.endswith("/") and fnmatch.fnmatch(
-                "/" + str(path.relative_to(self.base_path)), p + "*"
-            ):
-                return False
-            if (
-                p.endswith("/")
-                and p.startswith(str(self.base_path))
-                and fnmatch.fnmatch(str(path), p + "*")
-            ):
-                return False
-
-        return True
-
-    def _walk(self, this_path_str: str) -> Iterator[Entry]:
-        """
-        Walks path, returning an Entry iterator for each item.
-
-        If an item is not ignored, it is traversed recursively. Traversal stops on
-        ignored items.
-
-        Recalculates on every call.
-        """
-        debug_echo(f"Walking {this_path_str}")
-        this_path = Path(this_path_str)
-
-        # Handle non existent paths passed to cli.
-        # TODO handle further up
-        if not this_path.exists():
-            return
-
-        if this_path.is_file():
-            yield Entry(this_path, self._survives(this_path))
-        else:
-            for e in os.scandir(this_path_str):
-                if e.is_symlink():
-                    continue
-                elif this_path.match(LOG_FOLDER):
```

Comment:
Deleting this means we're going to now scan the `.semgrep_logs` folder that semgrep-action generates. I think we need to port this to semgrep. Make sure when testing to run on newly cloned repos, but also to run semgrep-action _twice_.

Reply:
Can we just add `.semgrep_logs` to the default `.semgrepignore`?

Reply:
ummmm sure

(Hesitance is because most users will probably always want to ignore it, but better to have it all in one place)

Reply:
added to default `.semgrep_ignore`